### ISSUE TITLE
Project Status

### ISSUE THREAD

**Issue by siavash-babaei** [mbraceproject/MBrace.Azure#173]
HI

Don't wanna step on any toes, however I would very much like to know whether the project is actively well-maintained and developed or not - seeing as there has been little/no activity/commint for over a year and AWS branch has been in experimental stage for quite a few years. 
It would be nice to hear some indication/statement as to the status and future of the project. 

THANX & MY-BEST

**Comment by isaacabraham** [comment]
@sashababaei Great question. Essentially the project is I would say in "hibernation". After .NET Core came out, the initial versions did not support many of the key features that were required by the core of MBrace - simultaneously, several key contributors to the project moved on.

My understanding is that many of the .NET Core issues have now been resolved and that a port to .NET Core would be doable (@dsyme and @eiriktsarpalis can probably comment further) although still a piece of work. @dsyme @eiriktsarpalis what would be the effort involved do you think in coming up with even a breakdown of what would actually be required to port the core elements (i.e. vagabond + mbrace core + cloud CE etc.) to .NET core?

Having said that, the existing version of MBrace works absolutely fine for .NET Framework (including the Azure part), although we've spoken about "trimming down" the number of libraries in the future.

**Comment by eiriktsarpalis** [comment]
Apart from the porting effort obviously, there's not much that can be done
before we can get our hands on a working release of fsi on .NET Core (iow
porting Vagabond).

**Comment by dsyme** [comment]
Scripting actually works for FSI on .NET Core today, you just have to pass a swathe of arguments

**Comment by dsyme** [comment]
I think https://github.com/Microsoft/visualfsharp/blob/master/tests/scripts/fsci.fsx gives a guide.  @KevinRansom might be able to advise

**Comment by sashababaei** [comment]
Thanx everyone for clearning things up. There is hope yet then ...

**Comment by dsyme** [comment]
BTW from an engineering perspective I think the only way to get efficient engineering progress for the MBrace collection of projects it to put them back in a single repo and single solution.

I tried making progress on some core engineering issues about 18 months ago but the split into multiple repos and nugets made pushing even a simple fix or design change through very difficult.

I appreciate that FsPickler needed to be moved out.  But based on my experience with the engineering of Fabulous I'd recommend that if/when we return to MBrace engineering we combine literally all of Vagabond, MBrace.Core, MBrace.Azure and MBrace.AWS into a single repo+solution with multiple output packages.  MBrace.Thespian should also likely be nuked or vastly simplified.

**Comment by isaacabraham** [comment]
Agreed - hence my comment regarding "trimming down" the libraries. As you say, push features through the plethora of packages was great in principle but was difficult in practice.

**Comment by sashababaei** [comment]
Thank you all for the great efforts and providing such wonderful tools. Just curious: Isn't it time for it to get out of hibernation ...?! Given that MS had a similar OneNet/Pirahna going, is it not possible to sell them the idea and secure funding/support so that a valuable project like this (or FsLab) could continuously and reliably be maintained?

**Comment by isaacabraham** [comment]
Hmmm. Maybe we should look for something on open collective ;-)

**Comment by isaacabraham** [comment]
@sashababaei you're right. And I still maintain that for ease of use, flexible and use cases MBrace provides an incredibly flexible framework for distributed computing - better than anything else out there on .NET IMHO.
